File: 485-MaxConsecutiveOnes/485-MaxConsecutiveOnes.java
/*

485. Max Consecutive Ones
Easy

Given a binary array, find the maximum number of consecutive 1s in this array.

Example 1:

Input: [1,1,0,1,1,1]
Output: 3
Explanation: The first two digits or the last three digits are consecutive 1s.
    The maximum number of consecutive 1s is 3.

Note:

    The input array will only contain 0 and 1.
    The length of input array is a positive integer and will not exceed 10,000

*/

class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int res = 0;
        int temp = 0;
        for(int n : nums){
            if (n == 1) temp++;
            else{
                res = Math.max(res, temp);
                temp = 0;
            }
        }
        return Math.max(res, temp);
    }
}

File: 485-MaxConsecutiveOnes/485-MaxConsecutiveOnes.py
"""

485. Max Consecutive Ones

Given a binary array nums, return the maximum number of consecutive 1's in the array.

Example 1:

Input: nums = [1,1,0,1,1,1]
Output: 3
Explanation: The first two digits or the last three digits are consecutive 1s. The maximum number of consecutive 1s is 3.

Example 2:

Input: nums = [1,0,1,1,0,1]
Output: 2

Constraints:

    1 <= nums.length <= 105
    nums[i] is either 0 or 1.

"""

class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        i = 0
        res = 0
        left = -1
        while i < len(nums):
            if nums[i] == 0: i+= 1
            else:
                left = i
                while i < len(nums) and nums[i] != 0: i += 1
                res = max(res, i - left)
        return res
